sqlmesh/core/snapshot/definition.py

```diff
@@ -1865,6 +1919,86 @@ def snapshots_to_dag(snapshots: t.Collection[Snapshot]) -> DAG[SnapshotId]:
     return dag
 
 
+def apply_auto_restatements(
+    snapshots: t.Dict[SnapshotId, Snapshot], execution_time: TimeLike
+) -> t.List[SnapshotIntervals]:
```

Comment:
can we check that this method doesn't have performance issues?, afraid of the cron calls etc